src/library_analyzer/processing/api/purity_analysis/_get_module_data.py

```diff
@@ -104,270 +108,315 @@ def _detect_scope(self, node: astroid.NodeNG) -> None:
+                self.targets = parent_targets
+
+        # Add all values that are used inside the function body to its values' dict.
+        if self.values:
+            for value in self.values:
+                # Do not add calls to value references
+                # if isinstance(value.node, MemberAccessValue) and value.node.member.attrname in self.current_function_def[-1].call_references:
+                #     continue
+                if self.find_first_parent_function(value.node) == self.current_function_def[-1].symbol.node:
+                    if value.name not in self.current_function_def[-1].value_references:
+                        self.current_function_def[-1].value_references[value.name] = [value]
+                    else:
+                        self.current_function_def[-1].value_references[value.name].append(value)
+            self.values = []
+
+    def _analyze_lambda(self, current_node: astroid.Lambda) -> None:
+        """Analyze a Lambda node.
+
+        This is called while the scope of a node is detected.
+        It must only be called when the current node is of type Lambda.
+        Add the Lambda FunctionScope to the functions' dict if the lambda function is assigned a name.
+        Add all values and calls that are collected inside the lambda to the Lambda FunctionScope instance.
+        Also add these values to the surrounding scope if it is of type FunctionScope.
+        This is due to the fact that lambda functions define a scope themselves
+        and otherwise the values would be lost.
+        """
+        if not isinstance(current_node, astroid.Lambda):
+            return
+
+        # Add lambda functions that are assigned to a name (and therefor are callable) to the functions' dict.
+        if isinstance(current_node, astroid.Lambda) and isinstance(current_node.parent, astroid.Assign):
+            # Make sure there is no AttributeError because of the inconsistent names in the astroid API.
+            if isinstance(current_node.parent.targets[0], astroid.AssignAttr):
+                node_name = current_node.parent.targets[0].attrname
+            else:
+                node_name = current_node.parent.targets[0].name
+            # If the Lambda function is assigned to a name, it can be called just as a normal function.
+            # Since Lambdas normally do not have names, they need to be assigned manually.
+            self.current_function_def[-1].symbol.name = node_name
+            self.current_function_def[-1].symbol.node.name = node_name
+            self.current_function_def[-1].symbol.id.name = node_name
+
+            # Extend the dict of functions with the current node or create a new list with the current node.
+            if node_name in self.functions:
+                self.functions[node_name].append(self.current_function_def[-1])
+            else:  # better for readability
+                self.functions[node_name] = [self.current_function_def[-1]]
```

Comment:
```suggestion
            self.functions.setdefault(node_name, []).append(self.current_function_def[-1])
```

Or use a [`defaultdict`](https://stackoverflow.com/a/4143719).

Reply:
The same pattern is applicable elsewhere.